### PR TITLE
server, sessionctx: workaround var initialization

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve? 

This provides a workaround for https://github.com/pingcap/tidb/issues/9382

I suggest a longer-term fix here: https://github.com/pingcap/tidb/issues/9381

### What is changed and how it works?

This makes various sysvars work:
- port
- datadir
- socket

Port was previously added in https://github.com/pingcap/tidb/pull/9365 , but it turns out this did not work correctly since it retrieved the 'default' port prior to parsing the config file.  I have changed the value to be "4000" in variables so it becomes less confusing since this doesn't work.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

The test-suite contains the server variables from the default config only, and so it is difficult to assert for these changes.

Code changes

 - Has exported variable/fields change

Side effects

 - Increased code complexity (workaround)

Related changes

 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: server/server.go [hunk 1]
```diff
@@ -182,6 +182,7 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 		stopListenerCh:    make(chan struct{}, 1),
 	}
 	s.loadTLSCertificates()
+	s.initDefaults()
```
Comment (jackysp) [inline server/server.go]: Or we could move it to `setGlobalVars`?

File: server/server.go [hunk 2]
```diff
@@ -229,6 +230,13 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 	return s, nil
 }
 
+func (s *Server) initDefaults() {
```
Comment (jackysp) [inline server/server.go]: It is better to add all https://github.com/pingcap/tidb/blob/master/tidb-server/main.go#L451 in this method.

**Comment by morgo** [comment]
@jackysp thanks for the suggestion, I've moved it.  PTAL again.

**Pull request merged**